sunpy/util/scraper.py

```diff
@@ -0,0 +1,189 @@
+        (i.e. using `~sunpy.time.TimeRange`).
+
+        Parameters
+        ----------
+
+        timerange : `~sunpy.time.TimeRange`
+            Time interval where to find the directories for a given
+            pattern.
+
+        Returns
+        -------
+
+        directories : list of strings
+            List of all the possible directories valid for the time
+            range given. Notice that these directories may not exist
+            in the archive.
+        """
+        #find directory structure - without file names
+        directorypattern = self.pattern[:-self.pattern[::-1].find('/')]
+        #TODO what if there's not slashes?
+        rangedelta = timerange.dt # TODO check it's a timerange
+        timestep = self._smallerPattern(directorypattern)
+        if timestep is None:
+            return [directorypattern]
+        else:
+            # Number of elements in the time range (including end)
+            n_steps = rangedelta.total_seconds()/timestep.total_seconds()
+            TotalTimeElements = int(round(n_steps)) + 1
+            directories = [(timerange.start + n * timestep).strftime(directorypattern)
+                        for n in range(TotalTimeElements)] #todo if date <= endate
+            return directories
+
+    def _URL_followsPattern(self, url):
+        """Check whether the url provided follows the pattern"""
+        time_conversions = {'%Y': '\d{4}', '%y': '\d{2}',
+                            '%b': '[A-Z]..', '%B': '\W', '%m': '\d{2}',
+                            '%d': '\d{2}', '%j': '\d{3}',
+                            '%H': '\d{2}', '%I': '\d{2}',
+                            '%M': '\d{2}',
+                            '%S': '\d{2}'}
+        pattern = self.pattern
+        for k,v in time_conversions.iteritems():
+            pattern = pattern.replace(k, v)
+        matches = re.match(pattern, url)
+        if matches:
+            return matches.end() == matches.endpos == len(self.now)
+        return False
+
+    def _extractDateURL(self, url):
+        """Extracts the date from a particular url following the pattern"""
+        url_to_list = lambda txt: re.sub(r'\.|_', '/', txt).split('/')
+        pattern_list = url_to_list(self.pattern)
+        url_list = url_to_list(url)
+
+        time_order = ['%Y', '%y', '%b', '%B', '%m', '%d', '%j',
+                      '%H', '%I', '%M', '%S']
+        final_date = []
+        final_pattern = []
+        #Find in directory and filename
+        for pattern_elem, url_elem in zip(pattern_list, url_list):
+            time_formats = [x for x in time_order if x in pattern_elem]
+            if len(time_formats) > 0:
+                final_date.append(url_elem)
+                final_pattern.append(pattern_elem)
+                for time_bit in time_formats:
+                    time_order.remove(time_bit)
+        return datetime.datetime.strptime(' '.join(final_date),
+                                          ' '.join(final_pattern))
```

Comment:
I have a url which goes like : http://stereo-ssc.nascom.nasa.gov/data/ins_data/plastic/level2/Protons/ASCII/10min/A/2008/STA_L2_PLA_1DMax_10min_20080101_001_V09.txt and hence cause group name redefinition error in return above. 

We get something like this:

final_pattern = ['%Y', '%Y%m%d', '%j']
final_date = ['2008', '20080101', '001']

So please change last return line to :

try:
    print datetime.datetime.strptime(' '.join(final_date), ' '.join(final_pattern))

except:
    anomaly = [x for x in final_pattern if len(x) > 2]
    for x in anomaly:
        splits = [x[i:i+2] for i in range(0, len(x), 2)]
        index_remove = [final_pattern.index(i) for i in splits if i in final_pattern]
        for i in splits:
            if i in final_pattern: final_pattern.remove(i)
                
        for i in index_remove: final_date.remove(final_date[i])
            

    print datetime.datetime.strptime(' '.join(final_date), ' '.join(final_pattern))